include/zephyr/net/http/service.h

```diff
@@ -220,14 +225,14 @@ struct http_service_desc {
  * @param _sec_tag_list_size TLS security tag list size used to setup a HTTPS socket.
  */
 #define HTTPS_SERVICE_DEFINE(_name, _host, _port, _concurrent, _backlog, _detail,              \
-			     _res_fallback, _sec_tag_list, _sec_tag_list_size)                 \
+			     _res_fallback, _sec_tag_list, _sec_tag_list_size, ...) \
```

Comment:
TBH I think I'd prefer to keep all of the macro parameters explicit instead of hiding optional parameters behind `__VA_ARGS__` - it might get super confusing if someone wanted to add yet another parameter in the future, it'd be tricky to document also.

We could just extend the macros/service struct with an additional `struct http_service_config` where we could keep any optional per-service configurations, including this custom socket struct, so in case any other custom configurations will be needed in the future, we could just extend the struct w/o touching the macros again.

Reply:
@rlubos
Good point!
If we want to both prevent future modifications and stay backwards compatible,
I can also alternatively create new macroses which accept `struct http_service_config`